guardrails/classes/history/outputs.py

```diff
@@ -82,23 +82,31 @@ def error_spans_in_output(self) -> List[ErrorSpan]:
 
         These indices are relative to the complete LLM output.
         """
-        total_len = 0
+        # map of total length to validator
+        total_len_by_validator = {}
         spans_in_output = []
         for log in self.validator_logs:
+            validator_name = log.validator_name
+            if total_len_by_validator.get(validator_name) is None:
+                total_len_by_validator[validator_name] = 0
```

Comment:
How does this work for structured data or other scenarios where the same validator may be used multiple times with different params